apps/api-nestjs/src/shared/vm-engine.service.ts

```diff
@@ -0,0 +1,147 @@
+import { HttpException, Injectable } from '@nestjs/common'
+import { ConfigService } from '@app/shared/config.service'
+import * as abi from 'ethereumjs-abi'
+import { TokenDto } from '@app/modules/token-transfers/dto/token.dto'
+import * as utils from 'web3-utils'
+import BigNumber from 'bignumber.js'
+import axios from 'axios'
+
+export interface VmEngineOptions {
+  rpcUrl: string
+  tokensAddress: VmContract
+}
+
+export interface VmContract {
+  address: string
+}
+
+/* tslint:disable:radix */
```

Comment:
@aldoborrero I had to disable the tslint radix rule for the vm-engine service because I'm pretty sure we need the default behaviour, but let me know if I should be supplying a radix param to the parseInt calls instead and I'll change it